### PR TITLE
Validation fix

### PR DESCRIPTION
**Pull request by shreeyash07**

### CODE REVIEW

File: manager-dashboard/app/utils/common.tsx [hunk 1]
```diff
@@ -104,9 +105,10 @@ export type IconKey = 'addOutline'
     | 'warningOutline'
     | 'generalTap'
     | 'tap'
-    | 'oneTap'
-    | 'twoTap'
-    | 'threeTap';
+    | 'tap-1'
+    | 'tap-2'
+    | 'tap-3'
+    | 'swipe-left';
```
Comment (tnagorra) [inline manager-dashboard/app/utils/common.tsx:111]: Let's make this consistent.
This should be either a kebab case or a camel case

File: manager-dashboard/app/views/NewTutorial/utils.ts [hunk 2]
```diff
@@ -651,13 +693,20 @@ export const tutorialFormSchema: TutorialFormSchema = {
                                         return undefined;
                                     }
 
+                                    const hasDuplicateSubOptionValue = checkDuplicate(
+                                        sub.map((val) => val.value).filter(isDefined),
+                                    );
```
Comment (tnagorra) [inline manager-dashboard/app/views/NewTutorial/utils.ts:698]: We do not need this validation as this is covered from another validation above.

File: manager-dashboard/app/views/NewTutorial/utils.ts [hunk 3]
```diff
@@ -599,19 +617,43 @@ export const tutorialFormSchema: TutorialFormSchema = {
...
+                        if (options.length > MAX_OPTIONS) {
                             return `There shouldn\`t be more than ${MAX_OPTIONS} options`;
                         }
-
+                        if (hasDuplicateOptionValue) {
+                            return 'Value cannot be same';
+                        }
+                        if (hasDuplicateAllOptionValue) {
+                            return 'Option and suboptions value cannot be same';
+                        }
```
Comment (tnagorra) [inline manager-dashboard/app/views/NewTutorial/utils.ts:656]: Let's also display which values are duplicated

File: manager-dashboard/app/views/NewTutorial/utils.ts [hunk 4]
```diff
@@ -434,6 +435,23 @@ export type PartialInformationPagesType = PartialTutorialFormType['informationPa
 export type PartialCustomOptionsType = PartialTutorialFormType['customOptions'];
 export type PartialBlocksType = NonNullable<NonNullable<PartialInformationPagesType>[number]>['blocks'];
 
+function checkDuplicate(array: Array<string | number>) {
```
Comment (tnagorra) [inline manager-dashboard/app/views/NewTutorial/utils.ts:438]: Let's use function from fujs

**Review by tnagorra — CHANGES_REQUESTED** [review]

File: manager-dashboard/app/views/NewTutorial/index.tsx [hunk 5]
```diff
@@ -350,16 +354,139 @@ function NewTutorial(props: Props) {
...
+                        tile_y: 'number',
+                        tile_z: 'number',
+                    }).join(', '),
+                ).filter(isTruthyString);
+
+                if (errors.length > 0) {
+                    return `Invalid GeoJson for Completeness: ${errors[0]} (${errors.length} total errors)`;
+                }
+            }
+            // TODO: validate references
```
Comment (tnagorra) [inline manager-dashboard/app/views/NewTutorial/index.tsx:461]: Let's also validate the screens and references.

The references should come from the custom options for footprint.
For others, the references are already hard-coded so we use 0, 1, 2, 3 (need to verify)

The screens should not be:
- duplicate (except for build area)
- start with 1
- should not have gaps

File: manager-dashboard/app/views/NewTutorial/index.tsx [hunk 6]
```diff
@@ -350,16 +354,139 @@ function NewTutorial(props: Props) {
     const handleGeoJsonFile = React.useCallback((
         geoProps: GeoJSON.GeoJSON | undefined,
     ) => {
-        const tutorialTasks = geoProps as PartialTutorialFormType['tutorialTasks'];
+        const tutorialTasks = geoProps as TutorialTasksGeoJSON;
+        function getGeoJSONError() {
```
[the remark this answers is not in the document]
Comment (tnagorra) [reply manager-dashboard/app/views/NewTutorial/index.tsx:358]: Done!

File: manager-dashboard/app/views/NewTutorial/index.tsx (hunk 5 above)
Comment (tnagorra) [reply manager-dashboard/app/views/NewTutorial/index.tsx:461]: From the discussion with @frozenhelium, let's show warning instead of error

**Review by frozenhelium — APPROVED** [review]

**Review by frozenhelium — APPROVED** [review]

**Pull request merged**